pkg/operator/ceph/pool/controller.go

```diff
@@ -90,10 +94,35 @@ func (c *PoolController) onAdd(obj interface{}) {
 		logger.Errorf("failed to get pool object: %+v", err)
 		return
 	}
-
+	if pool.Status == nil {
+		pool.Status = &cephv1.PoolStatus{}
+	}
+	if pool.Status.Phase != poolProcessingStatus {
+		pool.Status.Phase = poolProcessingStatus
```

Comment:
There is a lot of duplicated code here. How about a helper method that updates the status on the CR?

Reply:
yup i m working on it

Reply:
updated with a helper function for updating.